### DISCUSSION TITLE
Seqr installation on windows

### DISCUSSION THREAD

**Discussion by AhmadSafeer12345** [broadinstitute/seqr#4085] [Q&A]
Helo Team,

I am trying to install Seqr locally on a windows system. I looked over the instructions and it seems to be only for linux.  If you can guide about this, it would be grateful. 

Thanks!

**Comment by hanars** [comment]
Hi Ahmad,

We do not have any guidance for installing seqr on a windows operating system, it is not a supported OS. If you are able to get it to work we would love to hear from you what modifications you needed to made! However, it is probably better for you to run it on a linux VM.

**Comment by AhmadSafeer12345** [comment]
Helo,
Thank you for your response! I have locally installed Seqr on Linux and created a new project, but I didn't see anywhere I can upload my VCF files and start the analysis. Could you please let me know how I can proceed with that?

**Reply by hanars** [reply]
You can find instructions for how to load data here: https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#annotating-and-loading-vcf-callsets

**Comment by AhmadSafeer12345** [comment]
Hi @hanars , we have done all the steps and loaded the data in seqr, which you can see in the picture below:

![image](https://github.com/user-attachments/assets/5384ece0-f698-49e9-92fb-abeabe102774)


Now we try to load this data into our project as described here: [Adding dataset to seqr project](https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#adding-a-loaded-dataset-to-a-seqr-project)
We have now made the project and try to load this data as described above but get the  error which you can see in the screenshot below: 
![image](https://github.com/user-attachments/assets/46933c40-d64c-4e57-8f37-31a235dc53a0)
Now what could be the fix for this issue. Do we need to create the project in a specific way?  Kindly respond to this as soon as possible as this is hindering my workflow.

**Reply by hanars** [reply]
You need to create families and individuals in the project, using the "edit families and individuals" button

**Comment by AhmadSafeer12345** [comment]
Helo, I have created families and individuals as shown in the given image.
![Screenshot 2024-08-08 at 11-17-14 seqr BD](https://github.com/user-attachments/assets/41850c9c-017d-42ff-91ad-987f6dc1e17a)

Now, when I am trying to load data, it shows an error, as shown in the given image.
![Screenshot 2024-08-08 at 11-31-57 seqr BD](https://github.com/user-attachments/assets/f0b4152b-569c-40a8-9811-ace51f889b16)

what should I do?

**Reply by hanars** [reply]
You need every sample IDs in the the index you created to match an Individual IDs of the individuals you create. You need one individual for every sample in your VCF

**Reply by AhmadSafeer12345** [reply]
Helo again, I am trying to analyze only the index and have uploaded the vcf  file, created the project with families and individual. The project name and the index name in the elasticsearch status are the same.  when I try to load the dataset in the project through "Edit Datasets"   Again, it shows the same error " None of the individuals or samples in the project matched the 1 expected sample id(s). Could you please let me guide how to tackle this?

**Reply by hanars** [reply]
So in your VCF there are samples, each with a specific sample ID. Every sample in that VCF needs a corresponding Individual in seqr. On the project page in seqr you should see that you have "X families and Y individuals" - that number of individuals should match the number of samples in your VCF, and the Ids should match as well. Those individuals need to be created in the project before you can add the elasticsearch index. Once you have an individual in your project with the exact same ID as the sample in your VCF you can then load your data

**Reply by AhmadSafeer12345** [reply]
Hi,
We added the vcf file. However, now when we press submit, we get 401 Unauthorized error as shown in the screenshot below
![Screenshot from 2024-08-27 00-02-20](https://github.com/user-attachments/assets/80f9a558-61d4-4654-a6dc-4665a9886c2f)
Now on checking further details, I found out the error is related to not having a valid server token as given below:
```
{
    "error": "Postmark API response 401 (Unauthorized):\n{\n  \"ErrorCode\": 10,\n  \"Message\": \"Request does not contain a valid Server token.\"\n}"
}
```

Below if the curl command for the API so, you can further check it and let us know the issue.
```
curl 'http://localhost/api/project/R0008_120156436_1vcfgz/add_dataset/variants' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: text/plain;charset=UTF-8' \
  -H 'Cookie: accepted_cookies=true; sessionid=4zdrrsrq49d4l8r6v9wwmajlheam2br3; csrf_token=qNk7VzEZp68HRcuvEzukZc6R5r9T9skq' \
  -H 'Origin: http://localhost' \
  -H 'Referer: http://localhost/project/R0008_120156436_1vcfgz/project_page' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36' \
  -H 'X-CSRFToken: qNk7VzEZp68HRcuvEzukZc6R5r9T9skq' \
  -H 'sec-ch-ua: "Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  --data-raw '{"datasetType":"SNV_INDEL","elasticsearchIndex":"30358_120156_20a","ignoreExtraSamplesInCallset":true}'
```

**Reply by hanars** [reply]
This is a bug in seqr that we will fix, but that happens at the very end  of the adding the dataset and thereofre if you are seeing that error that means you have already succeeded in loading the data. If you just refresh the page you will see that the request succeeded and you now have search available

Also, for future reference, you should not be submitting this with the "Ignore extra samples in callset" checkbox selected. This means that you have done all the work to load data for the samples in the VCF and they are available in elasticsearch but you will **not** make them available in seqr. You should instead create families and individuals for every single sample in your VCF **before** you proceed with this final step to add the dataset

**Reply by AhmadSafeer12345** [reply]
Hi,

I have loaded the VCF file of an individual into seqr.  Now when I goes to variant search, it didn't proceed further and hanged on this link http://localhost/variant_search/family/F000005_436?

**Reply by AhmadSafeer12345** [reply]
It is showing the following errors

initialJSON:  
Object { meta: {…}, user: {…} }
project_page:1:1625
Creating store with initial state: 
Object { meta: {…}, user: {…}, projectsTableState: {…}, familyTableState: {…}, savedVariantTableState: {…}, variantSearchDisplay: {…}, searchesByHash: {}, familyTableFilterState: {…}, searchFamiliesByHash: {} }
app-aca8ceb6.js:1:1563546
No target defined 163-540b79f3.js:1:191023
TypeError: e is undefined
    ue http://localhost/static/180-ee0238f3.js:1
    value http://localhost/static/180-ee0238f3.js:1
    vl http://localhost/static/app-aca8ceb6.js:1
    Du http://localhost/static/app-aca8ceb6.js:1
    unstable_runWithPriority http://localhost/static/app-aca8ceb6.js:1
    Wo http://localhost/static/app-aca8ceb6.js:1
    ku http://localhost/static/app-aca8ceb6.js:1
    bu http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    unstable_runWithPriority http://localhost/static/app-aca8ceb6.js:1
    Wo http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    $o http://localhost/static/app-aca8ceb6.js:1
    Eu http://localhost/static/app-aca8ceb6.js:1
    notify http://localhost/static/app-aca8ceb6.js:1
    notifyNestedSubs http://localhost/static/app-aca8ceb6.js:1
    o http://localhost/static/app-aca8ceb6.js:1
    v http://localhost/static/app-aca8ceb6.js:1
    ZD http://localhost/static/app-aca8ceb6.js:1
    n http://localhost/static/app-aca8ceb6.js:1
    dispatch http://localhost/static/app-aca8ceb6.js:1
    b http://localhost/static/app-aca8ceb6.js:1
    handlePromise http://localhost/static/app-aca8ceb6.js:1
app-aca8ceb6.js:1:338859
pedigree image/png 163-540b79f3.js:1:140366
TypeError: Array(...).keys().map is not a function
    ee http://localhost/static/app-aca8ceb6.js:1
    la http://localhost/static/app-aca8ceb6.js:1
    qa http://localhost/static/app-aca8ceb6.js:1
    Va http://localhost/static/app-aca8ceb6.js:1
    Ma http://localhost/static/app-aca8ceb6.js:1
    Yl http://localhost/static/app-aca8ceb6.js:1
    Cu http://localhost/static/app-aca8ceb6.js:1
    Iu http://localhost/static/app-aca8ceb6.js:1
    Au http://localhost/static/app-aca8ceb6.js:1
    bu http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    unstable_runWithPriority http://localhost/static/app-aca8ceb6.js:1
    Wo http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    $o http://localhost/static/app-aca8ceb6.js:1
    Ne http://localhost/static/app-aca8ceb6.js:1
    Kt http://localhost/static/app-aca8ceb6.js:1
app-aca8ceb6.js:1:338859
Uncaught TypeError: Array(...).keys().map is not a function
    ee http://localhost/static/app-aca8ceb6.js:1
    la http://localhost/static/app-aca8ceb6.js:1
    qa http://localhost/static/app-aca8ceb6.js:1
    Va http://localhost/static/app-aca8ceb6.js:1
    Ma http://localhost/static/app-aca8ceb6.js:1
    Yl http://localhost/static/app-aca8ceb6.js:1
    Cu http://localhost/static/app-aca8ceb6.js:1
    Iu http://localhost/static/app-aca8ceb6.js:1
    Au http://localhost/static/app-aca8ceb6.js:1
    bu http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    unstable_runWithPriority http://localhost/static/app-aca8ceb6.js:1
    Wo http://localhost/static/app-aca8ceb6.js:1
    Ko http://localhost/static/app-aca8ceb6.js:1
    $o http://localhost/static/app-aca8ceb6.js:1
    Ne http://localhost/static/app-aca8ceb6.js:1
    Kt http://localhost/static/app-aca8ceb6.js:1
app-aca8ceb6.js:1:788529

**Reply by hanars** [reply]
Are you running that latest version of the seqr code? seqr does not auto-update so if you are seeing bugs you should always start by updating to the latest version:
https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#updating-seqr

**Reply by AhmadSafeer12345** [reply]
Hi,
I hope you are doing well. We updated seqr which was a hassle in itself. i had to stop my existing seqr container and then it made a new seqr container but now it doesn't have any of our data (.vcf files etc.). Did I do something wrong? Now I'll add the data again and see in the newer container created with the new version it it works correctly. Kindly let me know if I'll need to start a new container each time I update it.

**Reply by hanars** [reply]
I'm not sure what exactly you did to create a new container, but if you follow the instructions for updating seqr all your data should persist after the update going forward: https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#updating-seqr

**Reply by AhmadSafeer12345** [reply]
Hi,
I used the exact same instructions. However my existing container was named "ahmasafe-seqr-1". When I did 
`
docker-compose up -d
`
It started creating a new container named "seqr-seqr1" but failed to do so since the port was already allocated. My question is why did it create this new container when there was already one running there with my data? For this purpose now I had to stop the existing container "ahmasafe-seqr-1" and then again ran above command which created a new container but now since it was a new container and wasn't mapped to the existing volumes of the original one, hence it didn't contain any of the new data. Now how do I run the command with my existing container? This is a very weird kind of error.

**Reply by hanars** [reply]
I'm not really sure why your container was named `ahmasafe` in the first place - this sounds like you have changed your environment or where you were running these commands from in between runs. Essentially what you did was instead of updating your existing seqr deployment you created a different deployment in a different context. This is not something the seqr team can help you with, as making sure the environment and conditions under which you deploy seqr are maintained is your responsibility

**Reply by AhmadSafeer12345** [reply]
Yes, I have not managed to update the seqr version and its working fine now in the new version. I updated the paths for my custom usage in docker-compose and now able to get the old data as well. But just wanted to point out one thing. Whenever I run docker-compose up -d, it spins up the containers. However, for elastic search container it fails to run initially since it doesn't have the required permissions and I have to run the below command each time before to give it permission to access local directory
`
sudo chown -R 1000:1000 ./data/elasticsearch

`

my question is why do I have to give this permission whenever doing docker-compose up. Secondly is there a way to persist this? If not, I would suggest to add this to the documentation.

**Reply by hanars** [reply]
I think this issue is probably related to the permissions you have on your machine and those directories, not an issue within the seqr docker compose, but thats not something I can verify. seqr is an open source repository so if you think there are valuable updates to the documentation that others would benefit from you should feel free to put up a pull request with the updated documentation

**Reply by AhmadSafeer12345** [reply]
Helo Again,
 I have a family including parents and two members and uploaded each of their VCF file in Seqr. Now when I try to load  the data through dataset for the second individual in the same project it says "The following families are included in the callset but are missing some family members: 335 (15016_202926_22A). I have attached screen shots for your information. Could you please let me clarify this?  Does it require a single merge VCF files of each member? 
![Screenshot from 2024-10-29 11-22-40](https://github.com/user-attachments/assets/665ea874-5d35-444b-9042-ed5c520749b3)
![Screenshot from 2024-10-29 11-23-53](https://github.com/user-attachments/assets/132acb9c-d11c-4594-a92c-b55e373387f2)
![Screenshot from 2024-10-29 11-24-34](https://github.com/user-attachments/assets/e213a93f-2673-45c9-930d-8d6322d79d50)

**Comment by hanars** [comment]
You need to joint call all your data in the project together. To load more data you need to joint call your previously loaded data with the new samples you want to add. You need one VCF and one elasticsearch index per project

**Reply by AhmadSafeer12345** [reply]
Hello,
I have joined call  each individual and while uploading on seqr, it shows the given error.

Hail version: 0.2.122-be9d88a80695
Error summary: VCFParseError: missing value in FORMAT array. Import with argument 'array_elements_required=False'
DEBUG: 1 running tasks, waiting for next task to finish
INFO: Informed scheduler that task   SeqrVCFToMTTask__seqr_reference__VARIANTS__input_vcfs_Grow_c6d1b608eb   has status   FAILED
DEBUG: Asking scheduler for work...
DEBUG: Done
DEBUG: There are no more tasks to run at this time
DEBUG: There are 2 pending tasks possibly being run by other workers
DEBUG: There are 2 pending tasks unique to this worker
DEBUG: There are 2 pending tasks last scheduled by this worker
INFO: Worker Worker(salt=2147527017, workers=1, host=6b0c6b2eab45, username=root, pid=25) was stopped. Shutting down Keep-Alive thread
INFO: 
===== Luigi Execution Summary =====

Scheduled 3 tasks of which:
* 1 complete ones were encountered:
    - 1 VcfFile(filename=/input_vcfs/Growthmerged335/3351234.vcf.gz)
* 1 failed:
    - 1 SeqrVCFToMTTask(...)
* 1 were left pending, among these:
    * 1 had failed dependencies:
        - 1 SeqrMTToESTask(...)

This progress looks :( because there were failed tasks

===== Luigi Execution Summary =====


Could you please guide me as to how it can be solved?

**Reply by hanars** [reply]
This means you did not joint call the VCF in a way that is compatible with seqr. Please follow seqr's instructions for how to run a valid joint calling workflow and validating the expected output as defined here: https://storage.googleapis.com/seqr-reference-data/seqr-vcf-info.pdf
